prime-router/src/test/kotlin/fhirengine/azure/FHIRRouterIntegrationTests.kt

```diff
@@ -628,15 +626,19 @@ class FHIRRouterIntegrationTests : Logging {
         checkActionTable(listOf(TaskAction.convert, TaskAction.receive))
 
         // execute
-        var receivers = createReceivers(listOf(ReceiverSetupData("x", conditionFilter = observationFilter)))
+        var receiverSetupData = listOf(
+            ReceiverSetupData("x", conditionFilter = observationFilter),
+            ReceiverSetupData("y"),
```

Comment:
Ok, so I debugged this locally and for me at least there are a few of things going on here:

- receiver `x` isn't getting routed a bundle at all, the reason for that is actually due to how FHIR path works.  Your filter expression `Bundle.entry.resource.ofType(Observation).code.coding.code='94558-5'` retrieves every Observation and then grabs `code.coding.code` producing an array that looks like `[94558-5, 95418-0, 95417-2, 95421-4, 95419-8]` and `='94558-5'` is going to check that every item in the array matches
- The bundle that gets routed to receiver `y` has all five observations in it; that's occurring because the default you have configured in `ReceiverSetupData` is setting this `val conditionFilter: List<String> = listOf("true")`.  I would recommend removing all the defaults there and let the defaults in `Receiver` take effect; if you look in settings in stagings or prod we use an empty array if no filters are needed
  - A corollary of this is I think there is some additional clean up cases (we can defer this  to later) because the code actually has special handling when the filters are an empty array